### PR TITLE
Fixing Otel port derivation

### LINKED ISSUE #646: Parsing httpd endpoint in Otel-operator is broken.
I'm trying to use the httpd receiver with a declared endpoint, like : 

```
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: simplest
spec:
  config: |
    receivers:
      otlp:
        protocols:
          grpc:
            endpoint: localhost:9090
      httpd:
        endpoint: http://localhost:8080/server-status?auto
        collection_interval: 10s
    processors:
      nop:
    exporters:
      logging:
    service:
      pipelines:
        traces:
          receivers: [otlp]
          processors: []
          exporters: [logging]
        metrics:
          receivers: [httpd]
          processors: [nop]
          exporters: [logging]
```

However, I'm getting the following error once the CR is deployed: 
```
opentelemetry-operator-controller-manager-68cf456bd-6lmbs manager {"level":"info","ts":1640033774.9538267,"logger":"controllers.OpenTelemetryCollector","msg":"couldn't parse the endpoint's port","opentelemetrycollector":"opentelemetry-operator-system/simplest","endpoint":"http://localhost:80/server-status?auto"}
opentelemetry-operator-controller-manager-68cf456bd-6lmbs manager {"level":"info","ts":1640033774.9549108,"logger":"controllers.OpenTelemetryCollector","msg":"couldn't parse the endpoint's port","opentelemetrycollector":"opentelemetry-operator-system/simplest","endpoint":"http://localhost:80/server-status?auto"}
```

### PR DESCRIPTION
**Pull request by yuriolisa**
Signed-off-by: Yuri Sa <yurimsa@gmail.com>

- Fixing function portFromEndpoint to parse endpoint ports as expected.
- Fixed #646

### CODE REVIEW

File: pkg/collector/parser/receiver.go [hunk 1]
```diff
@@ -155,10 +157,19 @@ func portName(receiverName string, port int32) string {
 }
 
 func portFromEndpoint(endpoint string) (int32, error) {
-	i := strings.LastIndex(endpoint, ":") + 1
-	part := endpoint[i:]
-	port, err := strconv.ParseInt(part, 10, 32)
-	return int32(port), err
+	u, _ := url.Parse(endpoint)
+
+	host, port, _ := net.SplitHostPort(u.Host)
```
Comment (dmitryax) [inline pkg/collector/parser/receiver.go:162]: Please handle all the errors

File: pkg/collector/parser/receiver.go [hunk 2]
```diff
@@ -155,10 +157,19 @@ func portName(receiverName string, port int32) string {
...
+	
+	if len(host) > 0 {
+		port, err := strconv.ParseInt(port,10,32)
+		return int32(port), err 
+	} else {
```
Comment (dmitryax) [inline pkg/collector/parser/receiver.go:167]: else is not needed given you have the return above

**Review by dmitryax — COMMENTED** [review]
Please run go fmt and sign CLA

File: pkg/collector/parser/receiver.go (hunk 1 above)
Comment (yuriolisa) [reply pkg/collector/parser/receiver.go:162]: I'll do so.

File: pkg/collector/parser/receiver.go (hunk 2 above)
Comment (yuriolisa) [reply pkg/collector/parser/receiver.go:167]: ok, I got your point, however, I'm parsing the port variable in a different way on `else`.

**Review by VineethReddy02 — CHANGES_REQUESTED** [review]
@yuriolisa please adjust the go imports and format the go files.

**Pull request closed without merging**